### PR TITLE
Added navigation-react typings and tests

### PR DESCRIPTION
**Pull request by grahammendick**
- [x] Prefer to make your PR against the `types-2.0` branch.
- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code.
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped#common-mistakes).
- [x] Run `npm run lint -- package-name` if a `tslint.json` is present.

A new definition:
- [x] The package does not provide its own types, and you can not add them.
- [x] If this is for an NPM package, match the name. If not, do not conflict with the name of an NPM package.
- [x] Run `tsc` without errors.
- [x] Include the required [files](https://github.com/DefinitelyTyped/DefinitelyTyped#create-a-new-package) and header. Base these on the README, *not* on an existing project.

### CODE REVIEW

File: navigation-react/navigation-react-tests.tsx [hunk 1]
```diff
@@ -0,0 +1,79 @@
+import { StateNavigator } from 'navigation';
+import { NavigationBackLink, NavigationLink, RefreshLink } from '.';
```
Comment (andy-ms) [inline navigation-react/navigation-react-tests.tsx]: Please use a global import (`'navigation-react'`) instead.

File: navigation-react/navigation-react-tests.tsx [hunk 2]
```diff
@@ -0,0 +1,79 @@
...
+import * as React from 'react';
+
+namespace NavigationReactTests {
```
Comment (andy-ms) [inline navigation-react/navigation-react-tests.tsx]: Don't need this wrapper.

File: navigation-react/tsconfig.json [hunk 3]
```diff
@@ -0,0 +1,20 @@
+{
+    "compilerOptions": {
+        "module": "commonjs",
+        "target": "es6",
+        "noImplicitAny": true,
+        "strictNullChecks": false,
```
Comment (andy-ms) [inline navigation-react/tsconfig.json]: Why not?

**Comment by grahammendick** [comment]
Thanks for the review comments. I've implemented your proposed changes

**Pull request merged**